### ISSUE TITLE
Module '"@react-aria/utils"' has no exported member 'RouterProvider'.

### ISSUE THREAD

**Issue by wojtekmaj** [adobe/react-spectrum#5194]
### Provide a general summary of the issue here

Upon updating react-aria-components from 1.0.0-beta.0 to 1.0.0-beta.1, a TypeScript error appeared when type checking.

### 🤔 Expected Behavior?

No error I guess :shr

### 😯 Current Behavior

```
➤ YN0000: ../node_modules/react-aria/dist/types.d.ts(46,35): error TS2305: Module '"@react-aria/utils"' has no exported member 'RouterProvider'.
➤ YN0000: ../../node_modules/react-aria/dist/types.d.ts(46,35): error TS2305: Module '"@react-aria/utils"' has no exported member 'RouterProvider'.
```

### 💁 Possible Solution

I have no suggestions since I don't understand the change linked below, but I found a suspect commit:

https://github.com/adobe/react-spectrum/commit/15721e5a63ab036cb80e725a6c0001d628dce47c#diff-21c2686b609ef6206fc7f3bac09618071385feabe16b1f6fcff04014a78c4107L46

### 🖥️ Steps to Reproduce

* Install react-aria-components
* `tsc`

### Version

react-aria-components 1.0.0-beta.1

### What browsers are you seeing the problem on?

Other

### If other, please specify.

TypeScript check/build

### What operating system are you using?

macOS Sonoma

**Comment by reidbarber** [comment]
It does look like `RouterProvider` wasn't included in the types for some reason: https://cdn.jsdelivr.net/npm/@react-aria/utils@3.21.0/dist/types.d.ts.

Looking through the codebase, I'm not seeing many hints as to why. I see in `@react-spectrum/dropzone` we have a dep of `"@react-aria/utils": "^3.19.0"` instead of `^3.21.0"`, but I'm not sure that would have an effect.

**Comment by hzhu** [comment]
I'm using (just) react-aria and seeing this issue.

<img width="806" alt="Screenshot 2023-10-15 at 20 54 33" src="https://github.com/adobe/react-spectrum/assets/1811365/cf8a3d71-b2a9-4a7c-9997-4f2016170014">

**Comment by snowystinger** [comment]
@hzhu can you confirm your react-aria/utils version?

**Comment by snowystinger** [comment]
@wojtekmaj can you also confirm your installed version(s)? Sometimes package managers try to be smart and optimize for backend where it matters less if you have duplicate/multiple versions of a package and more that existing code paths stay the same as they were before. However, this is not the goal in frontend and many things will actually fail if you have multiple versions or display odd behavior.

If you don't know how to check this (different than the entry in your package.json) we can help if you tell us what package manager you're using (ie yarn/npm/pnpm)

**Comment by hzhu** [comment]
I'm on `"react-aria": "^3.29.0"`. 

Here's a [repository](https://github.com/hzhu/react-aria-minimal-example) with minimal code example (bootstrapped with `npx create-remix`. I added instructions on the `README.md`. Run `npx tsc` and the compile error will show up.

**Comment by snowystinger** [comment]
Thank you, I was able to verify you only have one copy.

I'm not sure why, but our build is treating this one `.tsx` file differently than our `.ts` files in this package and different from other `.tsx` files in the project. Looking into it more.

**Comment by hzhu** [comment]
Interesting 🤔. Thank you @snowystinger for looking into this 🙏🏽 .

**Comment by snowystinger** [comment]
@devongovett I could use your eyes on this as well. I've verified that our TS config can correctly run on this file by running
```
yarn tsc --project tsconfig.json  --declaration --emitDeclarationOnly
// with our tsconfig set to only run over
  "include": [
    "packages/@react-aria/utils/src/index.ts"
  ],
```
I also reduced all the other files in the package, so I'm only exporting from the problem file.

As noted above, if I get rid of the 'x' on the file extension (and just delete the jsx in the file), it also runs correctly.

Looking at the result emitted by ts in `transformer-typescript-types`, it's already wrong i think before it gets sent through ts. I was trying to verify that over here https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/js/src/JSTransformer.js but I have not understood how this hangs together well enough yet.

**Comment by wojtekmaj** [comment]
@snowystinger I'm getting @react-aria/utils@npm:3.21.0, exactly one copy.

**Comment by snowystinger** [comment]
Findings for why our automation didn't catch this. I'm not sure why, but our nextjs app actually has to import the RouterProvider in order for this issue to appear.
We also need to turn off the skipLibCheck, and we might want to copy the whole thing out to a different location because when we run typescript on our example apps, it actually uses types defined in node_modules our root project.
I'm not going to address this here, but I will open a ticket. Maybe we need a remix example app as well.

I've created a ticket for us here https://github.com/orgs/adobe/projects/19/views/4?pane=issue&itemId=41731571

**Comment by devongovett** [comment]
This is fixed in the `@react-aria/utils` v3.21.1 patch release, which is part of `react-aria` v3.29.1, `react-aria-components` v1.0.0-beta.2, and `@adobe/react-spectrum` v3.31.1 depending on how you are consuming it.
